tapcfg/server.go

```diff
@@ -326,8 +326,30 @@ func genServerConfig(cfg *Config, cfgLogger btclog.Logger,
 
 	multiNotifier := proof.NewMultiArchiveNotifier(assetStore, multiverse)
 
-	// TODO(ffranr): Replace the mock price oracle with a real one.
-	priceOracle := rfq.NewMockPriceOracle(3600, 5820600)
+	// Determine whether we should use the mock price oracle service or a
+	// real price oracle service.
+	var priceOracle rfq.PriceOracle
+
+	switch cfg.Experimental.Rfq.PriceOracleAddress {
+	case rfq.MockPriceOracleServiceAddress:
+		priceOracle = rfq.NewMockPriceOracle(
+			3600, cfg.Experimental.Rfq.MockOracleCentPerSat,
+		)
+
+	case "":
+		// Leave the price oracle as nil, which will cause the RFQ
```

Comment:
Because we allow the price oracle to be nil, we must expect users to run with the minimum flags they can get `litd` to run with. So it will very likely be `nil` for most users. So we need to remove the `nil` check in the `NewNegotiator` function, otherwise `litd` will not start:

```
// NewNegotiator creates a new quote negotiator.
func NewNegotiator(cfg NegotiatorCfg) (*Negotiator, error) {
	// If the price oracle is nil, then we will return an error.
	if cfg.PriceOracle == nil {
		return nil, fmt.Errorf("price oracle is nil")
	}
```